db/migrate/20230118102036_add_users_ref_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUsersRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```

Comment:
- [ ] The post should have an `author_id`, which means the reference should refer to `:author`, not `:user`. You can point Rails to the user table using the `to_table` option.

  You can look at [this StackOverflow answer](https://stackoverflow.com/a/42056089) on how to run the migration for this.